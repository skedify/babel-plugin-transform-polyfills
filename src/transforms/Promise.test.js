describe('Promise', () => {
  it('should replace new Promise with a polyfill', () => {
    const result = compile(
      `
      const c = new Promise(function (resolve, reject) {
        resolve({ a: 1 })
      })
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const e = new Promise(function (resolve, reject) {
        new Promise(function (resolve, reject) {
          setTimeout(resolve, 1000)
        }).then(resolve, reject)
      })
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const c = Object.keys(new Promise(function (resolve, reject) {
        resolve({ a: 1 })
      }))
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not touch other new expressions', () => {
    const result = compile(
      `
      const c = Object.keys(new Promise(function (resolve, reject) {
        new OtherClass(() => resolve({ a: 1 }))
      }))
    `
    )

    expect(result).toMatchSnapshot()
  })
})
