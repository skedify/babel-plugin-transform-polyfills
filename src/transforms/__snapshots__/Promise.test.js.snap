// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promise should not import polyfill more than once 1`] = `
"import _create_Promise from \\"babel-plugin-transform-polyfills/polyfills/Promise\\";

const e = _create_Promise(function (resolve, reject) {
  _create_Promise(function (resolve, reject) {
    setTimeout(resolve, 1000);
  }).then(resolve, reject);
});"
`;

exports[`Promise should not touch other new expressions 1`] = `
"import _create_Promise from \\"babel-plugin-transform-polyfills/polyfills/Promise\\";

const c = Object.keys(_create_Promise(function (resolve, reject) {
  new OtherClass(() => resolve({ a: 1 }));
}));"
`;

exports[`Promise should replace new Promise with a polyfill 1`] = `
"import _create_Promise from \\"babel-plugin-transform-polyfills/polyfills/Promise\\";

const c = _create_Promise(function (resolve, reject) {
  resolve({ a: 1 });
});"
`;

exports[`Promise should work when part of a call chain 1`] = `
"import _create_Promise from \\"babel-plugin-transform-polyfills/polyfills/Promise\\";

const c = Object.keys(_create_Promise(function (resolve, reject) {
  resolve({ a: 1 });
}));"
`;
